apollo-federation/src/operation/mod.rs

```diff
@@ -603,24 +606,33 @@ pub(crate) use selection_map::SelectionValue;
 /// * directives have to be applied in the same order
 /// * directive arguments order does not matter (they get automatically sorted by their names).
 /// * selection cannot specify @defer directive
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize)]
 pub(crate) enum SelectionKey {
     Field {
         /// The field alias (if specified) or field name in the resulting selection set.
         response_name: Name,
         /// directives applied on the field
+        // TODO(@TylerBloom): Add back in once apollo-compiler supports serde or a workaround is
```

Comment:
can you open a follow up JIRA to track this?

Reply:
Suggest using a `serde(serialize_with)` that just uses `directives.to_string()`?

Reply:
> Suggest using a `serde(serialize_with)` that just uses `directives.to_string()`?

That's a good idea. I'll add replace the relevant `serde(skip)`s will that.

Reply:
Here's the Jira ticket for the other `TODO`s.
https://apollographql.atlassian.net/browse/FED-363